block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockDagStorage.scala

```diff
@@ -64,6 +64,31 @@ final class InMemBlockDagStorage[F[_]: Concurrent: Sync: Log: BlockStore](
       _              <- lock.release
     } yield InMemBlockDagRepresentation(latestMessages, childMap, dataLookup, topoSort)
   override def insert(block: BlockMessage): F[Unit] =
+    genericInsert(block, normalLatestMessagesStrategy)
+  private def normalLatestMessagesStrategy(
+      block: BlockMessage,
+      latestMessages: Map[Validator, BlockHash]
+  ): Map[Validator, BlockHash] =
+    latestMessages.updated(block.sender, block.blockHash)
+  override def insertGenesis(genesis: BlockMessage): F[Unit] =
+    genericInsert(genesis, genesisLatestMessagesStrategy)
+  private def genesisLatestMessagesStrategy(
+      genesis: BlockMessage,
+      latestMessages: Map[Validator, BlockHash]
+  ): Map[Validator, BlockHash] = {
+    val genesisBonds = (for {
+      bd <- genesis.body
+      ps <- bd.state
+    } yield ps.bonds).getOrElse(List.empty[Bond])
```

Comment:
For some reason I couldn't import ProtoUtil.bonds

Reply:
You can't because block-storage doesn't depend on casper code

Reply:
You can use `BlockMessageUtil.bonds` though which will hopefully be removed once I get back to hiding optional fields from proto classes

Reply:
Yeah I'll leave it to you to do the refactoring, tests, and implementation for the non in memory version. If you don't see any other issues let's merge this for now to unblock the failing integration tests.